taichi/program.cpp

```diff
@@ -324,6 +341,15 @@ void Program::initialize_device_llvm_context() {
   }
 }
 
+Arch Program::get_snode_io_arch()
```

Comment:
Nit: s/io/rw/? Up to you, but i got confused a bit when i first saw it..

Reply:
I personally would vote for `rw` or `accessor`. `io` sounds like writing the SNodes to the hard disk :-)